cmd/detectExecuteScan.go

```diff
@@ -15,6 +17,18 @@ import (
 	"github.com/SAP/jenkins-library/pkg/versioning"
 )
 
+type utilsBundleDetectMaven struct {
+	*piperhttp.Client
+	*piperutils.Files
+}
+
+func newUtils() *utilsBundleDetectMaven {
+	return &utilsBundleDetectMaven{
+		Client: &piperhttp.Client{},
+		Files:  &piperutils.Files{},
+	}
+}
+
```

Comment:
It's really nice that you pick this up, but it is only used in line 152, yet, correct? The idea here is that only the function `detectExecuteScan()` would use `newUtils()`(name is too generic, btw, due to all steps sharing the same namespace in package cmd). All other functions would get the interface as a parameter. Then you can pass a mocking implementation in your tests. See also https://github.com/SAP/jenkins-library/pull/1913 for more context.